src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -8172,9 +8172,11 @@ StorageSnapshotPtr MergeTreeData::getStorageSnapshot(const StorageMetadataPtr &
 {
     auto snapshot_data = std::make_unique<SnapshotData>();
     ColumnsDescription object_columns_copy;
+    StreamSubscriptionPtr stream_subscription = nullptr;
 
     {
         auto lock = lockParts();
+        stream_subscription = subscribeForChanges();
```

Comment:
I double checked this, and no, the subscription is inside the storage snapshot, and it will only live at the  analyzer-planner stage, but it's still a good idea to propagate the flag to the getStorageSnapshot method.

please take a look at the structure StorageSnapshotSettings - added it to have another configurable place to create storage snapshots.